### PR TITLE
feat(ktable): setup syntax [khcp-6519]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

Switch to setup syntax.
Address [KHCP-6519](https://konghq.atlassian.net/browse/KHCP-6519).

## PR Checklist

* [x] Does not introduce dependencies
* [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [x] **Tests pass:** check the output of yarn test
* [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [x] **Framework style:** abides by the essential rules in Vue's style guide
* [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

[KHCP-6519]: https://konghq.atlassian.net/browse/KHCP-6519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/KTable/KTable.vue [hunk 1]
```diff
@@ -204,712 +204,656 @@ import KIcon from '@/components/KIcon/KIcon.vue'
...
+    offset.value = fetcherParams.offset
+    offsets.value.push(fetcherParams.offset)
+  }
 
-    const query = ref('')
-    const { debouncedFn: debouncedSearch, generateDebouncedFn: generateDebouncedSearch } = useDebounce((q: string) => { query.value = q }, 350)
-    const search = generateDebouncedSearch(0) // generate a debounced function with zero delay (immediate)
+  // get table headers
+  if (props.headers && props.headers.length) {
+    tableHeaders.value = props.headers as TableHeader[]
```
Comment (kaiarrowood) [inline src/components/KTable/KTable.vue:683]: Casting prop here to fix TS error

File: src/components/KTable/KTable.vue [hunk 2]
```diff
@@ -204,712 +204,656 @@ import KIcon from '@/components/KIcon/KIcon.vue'
...
+const tableId = computed((): string => props.testMode ? 'test-table-id-1234' : uuidv1())
+const defaultFetcherProps = {
+  pageSize: 15,
+  page: 1,
+  query: '',
+  sortColumnKey: '',
+  sortColumnOrder: 'desc',
+  offset: null,
+}
+const data = ref<[]>([])
```
Comment (kaiarrowood) [inline src/components/KTable/KTable.vue:508]: Adding `ref` type `<[]>` to allow usage in `defaultSorter` for `items` param

File: src/components/KTable/KTable.vue [hunk 3]
```diff
@@ -204,712 +204,656 @@ import KIcon from '@/components/KIcon/KIcon.vue'
...
+
+/**
+ * @deprecated defaultSorter
+ * @param {String} key - the current key to sort by
+ * @param {String} previousKey - the previous key used to sort by
+ * @param {String} sortOrder - either ascending or descending
+ * @param {Array} items - the list of items to sort
+ * @return {Object} an object containing the previousKey and sortOrder
+ */
+export const defaultSorter = (key: string, previousKey: string, sortOrder: string, items: []): Record<string, any> => {
```
Comment (kaiarrowood) [inline src/components/KTable/KTable.vue:852]:
```
(key: string, previousKey: string, sortOrder: string, items: [])
```

Initializing `items` to `[]` is requiring anything that is passed as `items` to need to be type `[]`

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**